### PR TITLE
Throw unexpected token error for } and > in JSXText nodes

### PR DESCRIPTION
**Pull request by kaicataldo**
Fixes https://github.com/acornjs/acorn-jsx/issues/106

Thoughts on the error message? The [corresponding PR to babel](https://github.com/babel/babel/pull/11046/files#diff-9b43fd1013c687b3088b5a4f72399cafR113-R119) also lists the HTML entity.

This is my first time contributing to this repository, so please let me know if I should be doing anything differently.

### CODE REVIEW

File: test/tests-jsx.js [hunk 1]
```diff
@@ -1461,316 +1461,316 @@ var fbTestFixture = {
...
-          "line": 1,
-          "column": 0
-        },
-        "end": {
-          "line": 1,
-          "column": 57
-        }
-      }
-    },
+    // '<LeftRight left=<a /> right=<b>monkeys /> gorillas</b> />': {
```
Comment (kaicataldo) [inline test/tests-jsx.js]: Thoughts on what to do with this? It looks like [this](https://github.com/RReverser/esprima-fb/blob/fb-harmony/test/fbtest.js#L1469) this is still in the test suite.

File: test/tests-jsx.js [hunk 2]
```diff
@@ -3958,3 +3958,657 @@ for (var ns in fbTestFixture) {
     });
   }
 }
+
+testFail("<LeftRight left=<a /> right=<b>monkeys /> gorillas</b> />", 'Unexpected token. Did you mean {">"} instead? (1:40)')
+testFail("<A>foo{</A>", "Unexpected token (1:8)")
```
Comment (kaicataldo) [inline test/tests-jsx.js]: These already throw an error.

File: test/tests-jsx.js [hunk 3]
```diff
@@ -3958,3 +3958,657 @@ for (var ns in fbTestFixture) {
...
+testFail("<A>foo<</A>", "Unexpected token (1:7)")
```
Comment (kaicataldo) [inline test/tests-jsx.js]: These already throw an error.

**Comment by RReverser** [comment]
Thanks for the PR.

I'm surprised this was not noticed for literally years since the change to spec was made... On one hand, it seems reasonable to fix it, but on another, I'm worried about breaking backwards compatibility in case there are projects that depend on this (and it seems all of the tools happily accepted these chars before).

**Comment by kaicataldo** [comment]
Thanks for the review!

I believe both Babel and TypeScript are adding this error, and ESLint is currently working on a major release in which we could add this breaking change. Another option on the ESLint side of things would be to check this in Espree, though I’m not sure of the performance implications of that, and it does feel to me like this should be in this plugin. 

Totally understand where you’re coming from, though! If this isn’t likely to land, I can explore the alternative I laid out above.

**Comment by RReverser** [comment]
@kaicataldo I wonder if it would make sense to:

1. Land this in Acorn JSX under an option (disabled by default).
2. Land this in major ESLint release with that option enabled by default.
3. Wait couple of months (?) until projects update.
4. Flip the default in major Acorn JSX release.

**Comment by kaicataldo** [comment]
Either of those options sounds good to me! What would you prefer?

File: test/tests-jsx.js (hunk 1 above)
Comment (kaicataldo) [reply test/tests-jsx.js]: I've updated this to include valid as well as invalid variants of this.

**Comment by kaicataldo** [comment]
Just wanted to follow up to see how you'd like to proceed. Thanks!

**Comment by RReverser** [comment]
I suppose we can even publish this as `5.2.0` - technically it's a minor breaking change, and semver-wise this would still prevent accidental automatic upgrade.

**Comment by kaicataldo** [comment]
That makes sense to me. Is there anything else you need from me before merging this?

**Comment by RReverser** [comment]
No, sorry, just been busy. I'll try to get to this on the weekend.

**Comment by kaicataldo** [comment]
No worries! Just wanted to make sure I wasn't holding anything up :)

**Comment by RReverser** [comment]
Apologies for the delay, and thanks again for the PR! Released as 5.2.0.

**Comment by kaicataldo** [comment]
No worries! Thanks for maintaining this project :)

**Comment by ljharb** [comment]
This minor release did, indeed, break eslint-plugin-react: https://github.com/yannickcr/eslint-plugin-react/pull/2583. There are no "minor" breaking changes :-(

**Comment by RReverser** [comment]
@ljharb I'm sorry for that :( This was a tough choice with unclear fallout due to it being, in theory, a bugfix to what should've never worked. I suppose unpublishing at this point might equally break users who started relying on the new version, so will have to leave as-is.

**Pull request merged**